plugins/restful/RestfulEntityBase.php

```diff
@@ -665,6 +735,15 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
     $save = FALSE;
     $original_request = $request;
 
+    $default_language = language_default();
+    $title_field = $this->getTitleField();
+
+    // Set the default language of a new instance if this bundle support
+    // field translation.
+    if ($this->getMethod() == \RestfulInterface::POST && entity_translation_node_supported_type($this->getBundle())) {
```

Comment:
> This helper class if for any kind of entity, will `entity_translation_node_supported_type` only work for nodes? Do we need a different helper function?

@e0ipso @amitaibu 
Is there any way to check translation support for other types?